io/pci/PowerVMEEH.py

```diff
@@ -83,21 +87,46 @@ def test_eeh_basic_pe(self):
-                self.log.info("Running error injection on pe %s function %s"
-                              % (self.addr, func))
-                self.return_code = self.basic_eeh(func, self.pci_class_name,
-                                                  self.pci_interface,
-                                                  self.pci_mem_addr,
-                                                  self.pci_mask)
-                if self.return_code == EEH_MISS:
-                    continue
-                else:
-                    if not self.check_eeh_pe_recovery(self.addr):
-                        self.fail("PE %s recovery failed after first EEH"
-                                  % self.addr)
+            while self.num_of_hit <= self.max_freeze:
+                for func in self.function:
+                    self.log.info("Running error inject on pe %s function %s"
+                                  % (self.addr, func))
+                    if self.num_of_miss < 5:
+                        self.return_code = self.basic_eeh(func,
+                                                          self.pci_class_name,
+                                                          self.pci_interface,
+                                                          self.pci_mem_addr,
+                                                          self.pci_mask)
+                        if self.return_code == EEH_MISS:
+                            self.num_of_miss += 1
+                            self.log.info("number of miss is %d"
+                                          % self.num_of_miss)
+                            continue
+                        else:
+                            self.num_of_hit += 1
+                            self.log.info("number of hit is %d"
+                                          % self.num_of_hit)
+                            if self.num_of_hit <= self.max_freeze:
+                                if not self.check_eeh_pe_recovery(self.addr):
+                                    self.fail("PE %s recovery failed after"
+                                              "%d EEH" % (self.addr,
+                                                          self.num_of_hit))
+                                    break
+                                else:
+                                    self.log.info("PE recovered successfully")
+                    else:
+                        self.log.warning("EEH inject failed for 5 times with\
+                                   function %s" % func)
+                        self.enter_loop = False
                         break
-                    self.log.info("PE recovered successfully")
+                if not self.enter_loop:
+                    break
+            if self.num_of_hit > self.max_freeze:
+                if self.check_eeh_removed():
+                    self.log.info("PE %s removed successfully" % self.addr)
+                    time.sleep(10)
```

Comment:
Discussed with @narasimhan-v , and thought its better to put sleep() in eeh_removed(), its better suited there.